products/views.py

```diff
@@ -1,3 +1,21 @@
-from django.shortcuts import render
+from django.views       import View
+from django.http        import JsonResponse
+from products.models    import MainCategory, SubCategory 
 
-# Create your views here.
+class CategoryListView(View):
+    def get(self, request):
+        results = [
+            {
+                "id"   : main.id,
+                "name" : main.name,
+                "sub_categories" : [
+                    {
+                        "id"   : sub.id,
+                        "name" : sub.name,
+                    }
+                    for sub in SubCategory.objects.filter(main_category_id=main.id)
+                ]
+            }
+            for main in MainCategory.objects.all() 
+        ]
```

Comment:
로직상 문제는 전혀없지만 이런식으로 조금 더 컴팩트하게 만들어줄 수 있을 것 같습니다~

```suggestion
        results = [{
            "id"   : main.id,
            "name" : main.name,
            "sub_categories" : [{
                "id"   : sub.id,
                "name" : sub.name,
            } for sub in SubCategory.objects.filter(main_category_id=main.id)]
        } for main in MainCategory.objects.all()]
```

Reply:
넵 훨씬 깔끔하네요! 반영했습니다